src/client/cl_predict.c

```diff
@@ -166,6 +171,8 @@ cm_trace_t Cl_Trace(const vec3_t start, const vec3_t end, const vec3_t mins, con
 	trace.skip = skip;
 	trace.contents = contents;
 
+	Cm_TraceBounds(start, end, mins, maxs, trace.box_maxs, trace.box_maxs);
+
 	Cl_ClipTraceToEntities(&trace);
```

Comment:
And finally, borrow the box intersection logic from the server so that we only clip traces to entities that reside within the trace's bounding box.